libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -172,11 +171,15 @@ func (d *addDockerMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		}
 
 		if len(container.Labels) > 0 {
-			labels := common.MapStr{}
+			//dedot first
+			outputLabels := common.MapStr{}
 			for k, v := range container.Labels {
-				safemapstr.Put(labels, k, v)
+				// This is necessary so that ES does not interpret '.' fields as new
+				// nested JSON objects, and also makes this compatible with ES 2.x.
```

Comment:
Is this planned to be backported? If not, I don't think we should have note inside for 2.x as 7.x is not expected to be compatible with 2.x

Reply:
definitely not for 2.x! thanks for the comments and I will remove it in next commit.